CONTRIBUTING.md

```diff
@@ -137,9 +144,143 @@ The [packages/](packages/) directory contains the following public/published pac
 ### Internal packages
 
 The [packages/](packages/) directory contains the following internal packages:
+
 * [caliper-publish](packages/caliper-publish/): Utility CLI for publishing Caliper to NPM and DockerHub.
 * [caliper-tests-integration](packages/caliper-tests-integration/): Collection of CI integration tests.
 
+### Testing Methodologies
+
+This section provides important information about testing this project. It includes dependencies used for testing and detailed instructions on how to add conforming tests.
+
+#### Testing Dependencies and Their Use
+
+* [mockery](https://www.npmjs.com/package/mockery): Mockery is a simple module for mocking Node.js modules during testing. It allows you to replace real modules with mocks or stubs.
+  
+* [mocha](https://mochajs.org/): Mocha is a feature-rich JavaScript test framework that runs on Node.js and in the browser. It facilitates asynchronous testing, making it easy to write simple and flexible tests.
+
+* [chai](https://www.npmjs.com/package/chai): Chai is a BDD/TDD assertion library for Node.js and the browser. It can be paired with any JavaScript testing framework. We use it to create readable and expressive assertions.
+
+* [sinon](https://sinonjs.org/releases/v18/): Sinon is a standalone test spies, stubs, and mocks for JavaScript. It works with any test framework and integrates well with Mocha and Chai. We utilize Sinon for checking how functions are called during testing.
+
+* [sinon-chai](https://www.npmjs.com/package/sinon-chai): This library provides a set of custom assertions for using Sinon with Chai. It allows you to write more readable assertions for Sinon spies, stubs, and mocks.
+
+* [nyc](https://www.npmjs.com/package/nyc): NYC is a command-line utility for generating code coverage reports. It is often used with Mocha to ensure that tests cover as much code as possible.
```

Comment:
I think we also need to include eslint and licence-check-and-add as these also form part of the unit testing pipeline and are mandatory. Unit tests for new code are also mandatory.

Reply:
done, i added it as a separate section to emphasize its importance